tools/c7n_azure/tests/test_container_host.py

```diff
@@ -0,0 +1,389 @@
+                        'type': 'container-event',
+                        'events': [{
+                            'resourceProvider': 'Microsoft.KeyVault/vaults',
+                            'event': 'write'
+                        }]
+                    }
+                })
+            }
+        }
+
+        # Verify we get all three events with no duplicates
+        host.update_event_subscriptions()
+        event_filter_mock.assert_called_with(key='Data.OperationName', values={
+            'Microsoft.KeyVault/vaults/write',
+            'Microsoft.Network/virtualNetworks/write',
+            'Microsoft.Resources/subscriptions/resourceGroups/write'})
+
+    @patch('c7n_azure.container_host.host.Host.has_required_params', return_value=True)
+    @patch('c7n_azure.container_host.host.BlockingScheduler.start')
+    @patch('c7n_azure.container_host.host.Host.prepare_queue_storage')
+    @patch('c7n_azure.container_host.host.Storage')
+    @patch('c7n_azure.container_host.host.Host.run_policies_for_event')
+    def test_poll_queue(self, run_policy_mock, storage_mock, _1, _2, _3):
+        host = Host()
+
+        host.policies = {
+            'one': {
+                'policy': ContainerHostTest.get_mock_policy({
+                    'name': 'one',
+                    'mode': {
+                        'type': 'container-event',
+                        'events': ['ResourceGroupWrite', 'VnetWrite']
+                    }
+                })
+            }
+        }
+
+        q1 = QueueMessage()
+        q1.id = 1
+        q1.dequeue_count = 0
+        q1.content = """eyAgCiAgICJzdWJqZWN0IjoiL3N1YnNjcmlwdGlvbnMvZWE5ODk3NGItNWQyYS00ZDk4LWE3
+        OGEtMzgyZjM3MTVkMDdlL3Jlc291cmNlR3JvdXBzL3Rlc3RfY29udGFpbmVyX21vZGUiLAogICAiZXZlbnRUeXBlIj
+        oiTWljcm9zb2Z0LlJlc291cmNlcy5SZXNvdXJjZVdyaXRlU3VjY2VzcyIsCiAgICJldmVudFRpbWUiOiIyMDE5LTA3
+        LTE2VDE4OjMwOjQzLjM1OTUyNTVaIiwKICAgImlkIjoiNjE5ZDI2NzQtYjM5Ni00MzU2LTk2MTktNmM1YTUyZmU0ZT
+        g4IiwKICAgImRhdGEiOnsgICAgICAgIAogICAgICAiY29ycmVsYXRpb25JZCI6IjdkZDVhNDc2LWUwNTItNDBlMi05
+        OWU0LWJiOTg1MmRjMWY4NiIsCiAgICAgICJyZXNvdXJjZVByb3ZpZGVyIjoiTWljcm9zb2Z0LlJlc291cmNlcyIsCi
+        AgICAgICJyZXNvdXJjZVVyaSI6Ii9zdWJzY3JpcHRpb25zL2VhOTg5NzRiLTVkMmEtNGQ5OC1hNzhhLTM4MmYzNzE1
+        ZDA3ZS9yZXNvdXJjZUdyb3Vwcy90ZXN0X2NvbnRhaW5lcl9tb2RlIiwKICAgICAgIm9wZXJhdGlvbk5hbWUiOiJNaW
+        Nyb3NvZnQuUmVzb3VyY2VzL3N1YnNjcmlwdGlvbnMvcmVzb3VyY2VHcm91cHMvd3JpdGUiLAogICAgICAic3RhdHVz
+        IjoiU3VjY2VlZGVkIiwKICAgfSwKICAgInRvcGljIjoiL3N1YnNjcmlwdGlvbnMvYWE5ODk3NGItNWQyYS00ZDk4LW
+        E3OGEtMzgyZjM3MTVkMDdlIgp9"""
+
+        q2 = QueueMessage()
+        q2.id = 2
+        q2.dequeue_count = 0
+        q2.content = q1.content
+
+        # Return 2 messages on first call, then none
+        storage_mock.get_queue_messages.side_effect = [[q1, q2], []]
+        host.poll_queue()
+        self.assertEqual(2, run_policy_mock.call_count)
+        run_policy_mock.reset_mock()
+
```

Comment:
Add `poll_queue` with `assertEqual(0)`? (To verify return 2 on first call then none)

Reply:
Oh poll queue is a while loop, so the fact that it is returning at all proves that it saw the empty array at the end of the side_effect.   If I called it again it would just fail unless we add more empty arrays to the mock.